ui/packages/lib/src/components/nav_drawer/NavDrawer.js

```diff
@@ -9,25 +9,56 @@ import {
   EuiHeaderSectionItemButton,
   EuiIcon,
   EuiFlexItem,
-  EuiSpacer
+  EuiSpacer,
+  EuiTreeView
 } from "@elastic/eui";
 import ApplicationsContext from "../../providers/application/context";
 import { CurrentProjectContext } from "../../providers/project";
-import { useToggle } from "../../hooks/useToggle";
+import { useToggle } from "../../hooks";
+import { navigate } from "@reach/router";
+import { get, slugify } from "../../utils";
+import urlJoin from "proper-url-join";
+
 import "./NavDrawer.scss";
 
 export const NavDrawer = ({ homeUrl = "/", docLinks }) => {
   const { projectId } = useContext(CurrentProjectContext);
   const { apps } = useContext(ApplicationsContext);
 
+  const isRootApplication = homeUrl === "/";
+
   const mlpLinks = useMemo(
     () =>
-      apps.map(a => ({
-        label: a.name,
-        iconType: a.icon,
-        href: projectId ? `${a.href}/projects/${projectId}` : a.href,
-        isActive: a.href === homeUrl
-      })),
+      apps.map(a => {
+        const isAppActive = a.href === homeUrl;
+
+        const children =
+          !!projectId && get(a, "config.sections")
+            ? a.config.sections.map(s => ({
+                id: slugify(`${a.name}.${s.name}`),
+                label: s.name,
+                callback: () =>
+                  navigate(urlJoin(a.href, "projects", projectId, s.href)),
+                className: "euiTreeView__node---small---subsection"
+              }))
+            : undefined;
+
+        return {
+          id: slugify(a.name),
+          label: a.name,
+          icon: <EuiIcon type={a.icon} />,
+          isExpanded: isAppActive || isRootApplication,
+          className: isAppActive
+            ? "euiTreeView__node---small---active"
+            : "euiTreeView__node---small",
+
+          callback: () =>
+            !children || !projectId
+              ? navigate(projectId ? `${a.href}/projects/${projectId}` : a.href)
+              : {},
+          children: children
+        };
+      }),
```

Comment:
`isRootApplication` is used as a dependency and raises a warning message. Either add it  as dependency, declare it inside the scope together with `isAppActive`, or add an ignore comment (`// eslint-disable-next-line react-hooks/exhaustive-deps`).